src/MPS.jl

```diff
@@ -612,7 +612,7 @@ function evolve!(::Canonical, ψ::AbstractAnsatz, mpo::AbstractMPO; threshold, m
     evolve!(NonCanonical(), ψ, mpo; threshold=nothing, maxdim=nothing, normalize=false, kwargs...) # set maxdim and threshold to nothing so we truncate from Canonical form
 
     if !isnothing(threshold) || !isnothing(maxdim)
-        truncate_sweep!(Canonical(), ψ; threshold, maxdim, normalize)
+        truncate_sweep!(Canonical(), ψ; threshold, maxdim, normalize, kwargs...)
```

Comment:
Also, you must modify the function signature here to not define `threshold`, `maxdim` and `normalize`.

```suggestion
        truncate_sweep!(Canonical(), ψ; kwargs...)
```